### DISCUSSION TITLE
lesson 6  foundry setup, can not generate lib folder

### DISCUSSION THREAD

**Discussion by zhongguoliliang** [Cyfrin/foundry-full-course-cu#1509] [General]
after enter forge init, can not generate lib folder like lesson 6, why ?Who can give me answer, thank you.
the first is lesson screenshot, the second is my code screenshot.
<img width="1086" alt="截屏2024-03-21 21 58 57" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/8157780/2a172772-4e63-423b-9a89-4c988a90d0b8">
<img width="986" alt="截屏2024-03-21 21 58 29" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/8157780/f5fefcd7-d710-4254-b079-617e66564e43">

**Comment by zhongguoliliang** [comment]
the error message is here
`git add exited with code 128:
warning: could not open directory 'Pictures/照片图库.photoslibrary/': Operation not permitted
warning: could not open directory 'Library/Application Support/MobileSync/': Operation not permitted
warning: could not open directory 'Library/Application Support/CallHistoryTransactions/': Operation not permitted
warning: could not open directory 'Library/Application Support/TotalAV/libs/avira/tmp/': Permission denied
warning: could not open directory 'Library/Application Support/com.apple.sharedfilelist/': Operation not permitted
warning: could not open directory 'Library/Application Support/Knowledge/': Operation not permitted
warning: could not open directory 'Library/Application Support/com.apple.TCC/': Operation not permitted
warning: could not open directory 'Library/Application Support/FileProvider/': Operation not permitted
warning: could not open directory 'Library/Application Support/AddressBook/': Operation not permitted
warning: could not open directory 'Library/Application Support/FaceTime/': Operation not permitted
warning: could not open directory 'Library/Application Support/com.apple.avfoundation/Frecents/': Operation not permitted
warning: could not open directory 'Library/Application Support/CallHistoryDB/': Operation not permitted
warning: could not open directory 'Library/Assistant/SiriVocabulary/': Operation not permitted
warning: could not open directory 'Library/Autosave Information/': Operation not permitted
warning: could not open directory 'Library/IdentityServices/': Operation not permitted
warning: could not open directory 'Library/Calendars/': Operation not permitted
warning: could not open directory 'Library/Messages/': Operation not permitted
warning: could not open directory 'Library/HomeKit/': Operation not permitted
warning: could not open directory 'Library/Sharing/': Operation not permitted
warning: could not open directory 'Library/com.apple.aiml.instrumentation/': Operation not permitted
warning: could not open directory 'Library/Mail/': Operation not permitted
warning: could not open directory 'Library/Trial/': Operation not permitted
warning: could not open directory 'Library/AppleMediaServices/': Operation not permitted
warning: could not open directory 'Library/DuetExpertCenter/': Operation not permitted
warning: could not open directory 'Library/Accounts/': Operation not permitted
warning: could not open directory 'Library/Safari/': Operation not permitted
warning: could not open directory 'Library/Biome/': Operation not permitted
warning: could not open directory 'Library/IntelligencePlatform/': Operation not permitted
warning: could not open directory 'Library/Shortcuts/': Operation not permitted
warning: could not open directory 'Library/Suggestions/': Operation not permitted
warning: could not open directory 'Library/Weather/': Operation not permitted
warning: could not open directory 'Library/Group Containers/group.com.apple.secure-control-center-preferences/': Operation not permitted
warning: could not open directory 'Library/Group Containers/group.com.apple.notes/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.VoiceMemos/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.archiveutility/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.Maps/Data/Maps/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.Home/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.Safari/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.CloudDocs.MobileDocumentsFileProvider/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.mail/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.MobileSMS/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.Notes/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.news/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.corerecents.recentsd/Data/Library/Recents/': Operation not permitted
warning: could not open directory 'Library/Containers/com.apple.stocks/': Operation not permitted
warning: could not open directory 'Library/PersonalizationPortrait/': Operation not permitted
warning: could not open directory 'Library/Photos/Libraries/Syndication.photoslibrary/': Operation not permitted
warning: could not open directory 'Library/Metadata/CoreSpotlight/': Operation not permitted
warning: could not open directory 'Library/Metadata/com.apple.IntelligentSuggestions/': Operation not permitted
warning: could not open directory 'Library/Cookies/': Operation not permitted
warning: could not open directory 'Library/CoreFollowUp/': Operation not permitted
warning: could not open directory 'Library/StatusKit/': Operation not permitted
warning: could not open directory 'Library/DoNotDisturb/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.HomeKit/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.safaridavclient/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.HomeKit.configurations/': Operation not permitted
warning: could not open directory 'Library/Caches/CloudKit/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.Safari/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.findmy.fmfcore/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.Safari.SafeBrowsing/': Operation not permitted
warning: could not open directory 'Library/Caches/FamilyCircle/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.homed/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.findmy.fmipcore/': Operation not permitted
warning: could not open directory 'Library/Caches/com.apple.ap.adprivacyd/': Operation not permitted
warning: could not open directory '.Trash/': Operation not permitted
warning: adding embedded git repository: .cocoapods/repos/master
hint: You've added another git repository inside your current repository.
hint: Clones of the outer repository will not contain the contents of
hint: the embedded repository and will not know how to obtain it.
hint: If you meant to add a submodule, use:
hint: 
hint:   git submodule add <url> .cocoapods/repos/master
hint: 
hint: If you added this path by mistake, you can remove it from the
hint: index with:
hint: 
hint:   git rm --cached .cocoapods/repos/master
hint: 
hint: See "git help submodule" for more information.
error: open(".rnd"): Permission denied
error: unable to index file '.rnd'
fatal: adding files failed

**Comment by EngrPips** [comment]
Hello @zhongguoliliang , I can see lib folder your codebase repo, so how do you mean you couldn't generate lib folder like lesson 6 ?

**Reply by zhongguoliliang** [reply]
Hello，@EngrPips,the first picture you see is the screenshot of the online lesson, the second one is my code screenshot. I upload my code below.
<img width="986" alt="截屏2024-03-21 21 58 29" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/8157780/e09e8d64-e8ba-481a-877e-e90f8c3b043f">

**Reply by EngrPips** [reply]
Hello @zhongguoliliang , Do you know if you have administrative controls in the shell ? , try open a cmd in administrator mode and let's see if that goes through.

**Reply by zhongguoliliang** [reply]
Thank you for your adverse, but it still not work.

**Reply by EngrPips** [reply]
Do you confirm if you have administrative controls in the shell ?

**Reply by zhongguoliliang** [reply]
yes, I have the root on Mac .Maybe I need change to a Windows PC!

**Reply by EngrPips** [reply]
Setting up stuffs can be the hardest part sometimes

**Reply by zhongguoliliang** [reply]
Yes,I will try on windows, if I still meet problem，I will come here ask you, thank you very much  EngrPips!

**Reply by EngrPips** [reply]
Alright no problem, i hope you don't meet problem anymore lol

**Reply by zhongguoliliang** [reply]
Hai,EngrPips. Are you still there ！I try it on windows, but the same problem happened！You can see the picture below! I run the visual studio code as administrator, what the problem!😥Thanks.
![IMG_2610](https://github.com/Cyfrin/foundry-full-course-f23/assets/8157780/773f0a00-1c4f-4681-96e5-d05319e3c170)

**Reply by EngrPips** [reply]
Yeah i am here friend, you are still having issue with the lib folder ? , please run the below command in your terminal 

`forge install foundry-rs/forge-std` 

lets see if that automatically populate the lib folder.

**Reply by zhongguoliliang** [reply]
Ok， I will have a try! Thank you very much!

**Reply by EngrPips** [reply]
Alright, Looking forward to hear from you

**Reply by zhongguoliliang** [reply]
It works, EngrPips, thank you very much! Do you have social app like YouTube Twitter WhatsApp or something else!I want to add you as friend and ask question directly! Thank you very much again!

**Reply by EngrPips** [reply]
You sure my twitter is @EngrPips  you can add me up. i use discord too if that is a thing for you, my discord is engr.pips

**Reply by zhongguoliliang** [reply]
Ok, I have followed you on twitter and add you on discord!

**Reply by EngrPips** [reply]
Awesome buddy, i accepted friend request on discord and i would add you back on twitter in a bit.

**Reply by EngrPips** [reply]
Also i see you are not cyfrin discord server, please join with this link `https://discord.gg/J8SgPrxQ` if you will.

**Reply by zhongguoliliang** [reply]
OK，I have join the group！Thank you！

**Reply by EngrPips** [reply]
Awesome, so nice to have you onboard.
